### PR TITLE
Cosmos Spark: Changing inferSchema.forceNullableProperties default to true

### PR DESCRIPTION
**Pull request by ealsur**
Based on user experience, we determined that changing the default of `spark.cosmos.read.inferSchema.forceNullableProperties` to have all properties nullable by default would be a beneficial starting experience, instead of having it to false and facing scenarios where the sampled documents contained the property but the rest of the documents do not, creating an exception during Json to Row conversion due to incompatible schema.

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
Thanks!

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala [hunk 1]
```diff
@@ -460,6 +460,57 @@ class SparkE2EQueryITest
     fieldNames.contains(CosmosTableSchemaInferrer.AttachmentsAttributeName) shouldBe false
   }
 
+  "spark query" can "when forceNullableProperties is false and rows have different schema" in {
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala:463]: We should also keep/add a test with forceNullableProperties == true. Basically the one where you figured out that we generate a less than ideal query when using a nested property.

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala:463]: Not necessarily blocking this PR - can happen seperately

Comment (ealsur) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITest.scala:463]: Added new test for explain when its true

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ When doing read operations, users can specify a custom schema or allow the conne
...
 | `spark.cosmos.read.inferSchema.includeTimestamp`     | `false`    | When schema inference is enabled, whether the resulting schema will include the document Timestamp (`_ts`). Not required if `spark.cosmos.read.inferSchema.includeSystemProperties` is enabled, as it will already include all system properties. |
-| `spark.cosmos.read.inferSchema.forceNullableProperties`     | `false`    | When schema inference is enabled, whether the resulting schema will make all columns nullable. By default whether inferred columns are treated as nullable or not will depend on whether any record in the sample set has null-values within a column. If set to `true` all columns will be treated as nullable even if all rows within the sample set have non-null values. |
+| `spark.cosmos.read.inferSchema.forceNullableProperties`     | `true`    | When schema inference is enabled, whether the resulting schema will make all columns nullable. By default, all columns will be treated as nullable even if all rows within the sample set have non-null values. When disabled, the inferred columns are treated as nullable or not depending on whether any record in the sample set has null-values within a column.  |
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:51]: should we mention here, how system properties are treated? here. all columns doesn't include system properties.

perhaps change
By default, **all columns** will be treated as nullable even if all rows within the sample set have non-null values.

to something like
By default, all columns (except cosmos system properties) will be treated as nullable even if all rows within the sample set have non-null values.

Comment (moderakh) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:51]: please consider adding a relevant changelog entry for this.

**Review by moderakh — APPROVED** [review]
LGTM.

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md (hunk 2 above)
Comment (ealsur) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:51]: Done

Comment (ealsur) [reply sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:51]: Done

**Pull request merged**